File: baker-test/src/test/scala/com/ing/baker/test/javadsl/BakerEventsFlowTest.scala
package com.ing.baker.test.javadsl

import org.scalatest.matchers.should.Matchers
import org.scalatest.flatspec.AnyFlatSpec

class BakerEventsFlowTest extends AnyFlatSpec with Matchers {
  "BakerEventsFlow object" should "be created" in {
    BakerEventsFlow.of("a", "b", "c")
    BakerEventsFlow.ofClass(classOf[Object])
  }

  "BakerEventsFlow object" should "be test for equals" in {
    val result = BakerEventsFlow.of("Object", "BakerEventsFlowTest") ==
      BakerEventsFlow.ofClass(classOf[Object], classOf[BakerEventsFlowTest])
    assertResult(true)(result)
  }

  "BakerEventsFlow object" should "be test for equals and order should not matter" in {
    val result = BakerEventsFlow.of("BakerEventsFlowTest", "Object") ==
      BakerEventsFlow.ofClass(classOf[Object], classOf[BakerEventsFlowTest])
    assertResult(true)(result)
  }
}

File: baker-test/src/test/java/com/ing/baker/test/javadsl/BakerEventsFlowTest.java
package com.ing.baker.test.javadsl;

import org.junit.Assert;
import org.junit.Test;

public class BakerEventsFlowTest {
    @Test
    public void testClassVsString() {
        Assert.assertEquals(BakerEventsFlow.of("Object"), BakerEventsFlow.ofClass(Object.class));
    }

    @Test
    public void testOrderDoesNotMatter() {
        Assert.assertEquals(BakerEventsFlow.of("a", "b", "c"), BakerEventsFlow.of("c", "a", "b"));
    }
}
